src/crypto.rs

```diff
@@ -1,7 +1,165 @@
+use thiserror::Error;
 use ssz_rs::prelude::*;
+use rand::prelude::*;
+
+use blst::{min_pk as blst_core, BLST_ERROR};
+use crate::crypto::Error::SizeMismatch;
+use blst::min_pk::AggregateSignature;
+
+pub const BLS_PUBLIC_KEY_BYTES_LEN: usize = 48;
+pub const BLS_SECRET_KEY_BYTES_LEN: usize = 32;
+pub const BLS_SIGNATURE_BYTES_LEN: usize = 96;
+
+pub type BLSPubkey = Vector<u8, BLS_PUBLIC_KEY_BYTES_LEN>;
+pub type BLSSignature = Vector<u8, BLS_SIGNATURE_BYTES_LEN>;
```

Comment:
will want to remove these once we decide how to unify the below representations with the SSZ functionality

Reply:
ok